lark/lexer.py

```diff
@@ -76,14 +76,15 @@ def max_width(self):
 
 
 class TerminalDef(Serialize):
-    __serialize_fields__ = 'name', 'pattern', 'priority'
+    __serialize_fields__ = 'name', 'pattern', 'priority', 'nice_print'
```

Comment:
Why have a `nice_print` attribute?

I think it's better to have it as a method that returns a different result based if `self.pattern` is `PatternStr` and `PatternRE`

Also, you can just call it `user_repr()`

Reply:
My plan is to improve this to actually contain the raw source code as accessible in `load_grammar`. (and line information etc.) This information isn't being carried around at the moment, and since it doesn't need to be used for anything but this I decided to just generate the string early. But maybe `user_repr` is a better name. `nice_print` is the best I could come up with.

Reply:
That could be nice. And then instead of `all_terminals` we can have a more generic term like `grammar_info`

Reply:
Do you mean that I just rename `all_terminals` to `grammar_info` without changing the content? Or do you want to put more info in the attribute?

Reply:
You just said you want to put more content. So yeah, I meant more content, not just rename.

Reply:
Well, I didn't plan to add more content to the Exception classes `_all_terminals` attribute. Just adding more content to TerminalDef.user_repr.